tests/test_data/custom_bleu/custom_bleu_for_language_generation.py

```diff
@@ -0,0 +1,262 @@
+machine-translated from one natural language to another. Quality is considered to be the correspondence between a 
+machine's output and that of a human: "the closer a machine translation is to a professional human translation, the 
+better it is" – this is the central idea behind BLEU. BLEU was one of the first metrics to claim a high correlation 
+with human judgements of quality, and remains one of the most popular automated and inexpensive metrics.
+
+Scores are calculated for individual translated segments—generally sentences—by comparing them with a set of good 
+quality reference translations. Those scores are then averaged over the whole corpus to reach an estimate of the 
+translation's overall quality. Intelligibility or grammatical correctness are not taken into account[citation needed].
+
+BLEU's output is always a number between 0 and 1. This value indicates how similar the candidate text is to 
+the reference texts, with values closer to 1 representing more similar texts. Few human translations will attain a 
+score of 1, since this would indicate that the candidate is identical to one of the reference translations. For this 
+reason, it is not necessary to attain a score of 1. Because there are more opportunities to match, adding additional
+reference translations will increase the BLEU score.
+"""
+
+_KWARGS_DESCRIPTION = """
+Computes BLEU score of translated segments against one or more references.
+Args:
+    predictions: list of translations to score.
+        Each translation should be tokenized into a list of tokens.
+    references: list of lists of references for each translation.
+        Each reference should be tokenized into a list of tokens.
+    max_order: Maximum n-gram order to use when computing BLEU score.
+    smooth: Whether or not to apply Lin et al. 2004 smoothing.
+Returns:
+    'score': BLEU score,
+    'precisions': geometric mean of n-gram precisions,
+    'brevity_penalty': brevity penalty,
+    'length_ratio': ratio of lengths,
+    'translation_length': translation_length,
+    'reference_length': reference_length
+Examples:
+
+    >>> bleu = jury.load_metric("bleu")
+    >>> predictions = [["the cat is on the mat", "There is cat playing on the mat"], ["Look! a wonderful day."]]
+    >>> references = [
+        ["the cat is playing on the mat.", "The cat plays on the mat."], 
+        ["Today is a wonderful day", "The weather outside is wonderful."]
+    ]
+    >>> results = bleu.compute(predictions=predictions, references=references)
+    >>> print(results)
+    {'bleu': {'score': 0.42370250917168295, 
+        'precisions': [0.8823529411764706, 0.6428571428571429, 0.45454545454545453, 0.125], 
+        'brevity_penalty': 1.0, 'length_ratio': 1.0, 'translation_length': 11, 'reference_length': 11}}
+"""
+
+
+@datasets.utils.file_utils.add_start_docstrings(_DESCRIPTION, _KWARGS_DESCRIPTION)
+class CustomBleuForLanguageGeneration(MetricForLanguageGeneration):
```

Comment:
What does custom bleu means?

Reply:
This is solely for testing purposes. The implementation here is duplicate of bleu it's exactly the same, however this custom_bleu is used to test the viability of custom metrics outside jury.